asyncpg/connection.py

```diff
@@ -1272,6 +1275,17 @@ def _drop_global_statement_cache(self):
                     after = time.monotonic()
                     timeout -= after - before
 
+        except exceptions.InternalClientError:
```

Comment:
I'd like to see a specific exception being used here, like `OutdatedTypeCacheError`.  It can be derived from `InternalClientError` to make it easier to write code that just want to handle a broad set of errors.